CompanyEmployees.Infrastructure.Persistence/Repositories/AttendanceRepository.cs

```diff
@@ -0,0 +1,44 @@
+using CompanyEmployees.Core.Domain.Entities;
+using CompanyEmployees.Core.Domain.Repositories;
+using Microsoft.EntityFrameworkCore;
+
+namespace CompanyEmployees.Infrastructure.Persistence.Repositories;
+
+internal sealed class AttendanceRepository : RepositoryBase<Attendance>, IAttendanceRepository
+{
+    public AttendanceRepository(RepositoryContext repositoryContext)
+        : base(repositoryContext)
+    {
+    }
+
+    public async Task<IEnumerable<Attendance>> GetAllAttendancesAsync(bool trackChanges, CancellationToken ct = default) =>
+        await FindAll(trackChanges)
+            .OrderByDescending(a => a.CheckInTime)
+            .ToListAsync(ct);
+
+    public async Task<Attendance> GetAttendanceAsync(Guid attendanceId, bool trackChanges, CancellationToken ct = default) =>
+        await FindByCondition(a => a.Id.Equals(attendanceId), trackChanges)
+            .SingleOrDefaultAsync(ct);
+
+    public async Task<IEnumerable<Attendance>> GetAttendancesByEmployeeIdAsync(Guid employeeId, bool trackChanges, CancellationToken ct = default) =>
+        await FindByCondition(a => a.EmployeeId.Equals(employeeId), trackChanges)
+            .OrderByDescending(a => a.CheckInTime)
+            .ToListAsync(ct);
+
+    public async Task<Attendance?> GetTodayAttendanceByEmployeeIdAsync(Guid employeeId, bool trackChanges, CancellationToken ct = default)
+    {
+        // BUG: Using DateTime.Now instead of DateTime.UtcNow - timezone issue
+        var today = DateTime.Now.Date;
```

Comment:
Using `DateTime.Now` can lead to inconsistent behavior across time zones. It's safer to use `DateTime.UtcNow` for server-side filtering to avoid local time discrepancies.

```suggestion
var today = DateTime.UtcNow.Date;
```